sklearn/base.py

```diff
@@ -377,6 +378,44 @@ def _check_n_features(self, X, reset):
                                        self.n_features_in_)
                 )
 
+    def _check_feature_names(self, X, reset=True):
+        """Validate feature names and set or check the `feature_names_in_`
+        attribute.
+
+        Parameters
+        ----------
+        X : {dataframe-like} of shape (n_samples, n_features)
+            The input samples.
+        reset : bool, default=True
+            Whether to reset the `feature_names_in_` attribute.
+            If False, the Input will be checked for consistency with data
+            provided when reset was last True.
+        """
```

Comment:
do we want to add sth like "Most estimators would call this method with `reset=True` in `fit` and `reset=False` in other methods." for give a hint to third party devs?